runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -7727,11 +7729,13 @@ void J9::Power::TreeEvaluator::generateCheckForValueTypeMonitorEnterOrExit(TR::N
    TR_J9VMBase *fej9 = (TR_J9VMBase *)(cg->fe());
    TR::MemoryReference *classFlagsMemRef = TR::MemoryReference::createWithDisplacement(cg, objectClassReg, static_cast<uintptr_t>(fej9->getOffsetOfClassFlags()), 4);
 
-   //check J9ClassIsValueType flag
-   generateTrg1MemInstruction(cg,TR::InstOpCode::lwz, node, tempReg, classFlagsMemRef);
-   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::andi_r, node, tempReg, tempReg, condReg, J9ClassIsValueType);
+   //check J9_CLASS_DISALLOWS_LOCKING_FLAGS (J9ClassIsValueType | J9ClassIsValueBased)
+   generateTrg1MemInstruction(cg, TR::InstOpCode::lwz, node, temp1Reg, classFlagsMemRef);
 
-   //If obj is value type, call VM helper and throw IllegalMonitorState exception, else continue as usual
+   loadConstant(cg, node, classFlag, temp2Reg);
```

Comment:
The fact that the new flag is outside of the immediate value range for Power is a bit concerning but I guess we ran out of smaller flags? @tajila

Reply:
Yes, the smaller flags are all used up. How large is the immediate range on power?

Reply:
@tajila it’s 16 bits.